lib/attr_json/record/dirty.rb

```diff
@@ -137,20 +137,20 @@ def saved_change_to_attribute?(attr_name)
         end
 
         def saved_changes
-          saved_changes = model.saved_changes
-          return {} if saved_changes == {}
+          original_saved_changes = model.saved_changes
+          return {} if original_saved_changes == {}
 
           json_attr_changes = registry.definitions.collect do |definition|
-            if container_change = saved_changes[definition.container_attribute]
-              old_v = container_change[0][definition.store_key]
-              new_v = container_change[1][definition.store_key]
+            if container_change = original_saved_changes[definition.container_attribute]
+              old_v = container_change.dig(0, definition.store_key)
```

Comment:
Using `#dig` will limit us to Ruby 2.3+. Which might be fine, though 2.2 is not yet EOL'd. Also, we should adjust the `.gemspec` to specify the minimum Ruby version is 2.3, in that case. (This same comment goes for the usages below, too.)

Reply:
hmm. As a result of other changes in PR, I do need to change it to the semantics of dig. I could write it myself as an expression. Currently travis is only running on 2.4 and 2.5. 

What do you think the right lower point to support is? I'm not sure. Convenient to use new rubies, and _I'm_ planning on using this gem with greenfield projects, you?

Reply:
It looks like maybe 2.2 is in fact already EOL? This page both says that it will be EOL as of March 31 2018 and that it's still under security maintenance; I think the page hasn't been updated since March 31?  2.3 is still in support. 

https://www.ruby-lang.org/en/downloads/branches/

So I think either support 2.3+ or 2.4+, no reason to support 2.2.  Even 2.4+ is fine with me -- which seems to be de facto what i've been doing, as travis tests only on 2.4+.

Reply:
> So I think either support 2.3+ or 2.4+ (either way we get dig), no reason to support 2.2.

Sounds fine to me, but we should update the `.gemspec` to reflect that. And probably mention it in the `README` too. I'm using this on a two year old application, but we're running Ruby 2.5.1 and Rails 5.2.0, so...

Reply:
ok, i'm going to go with 2.4+ then. i'm not used to gemspec having ruby dependency, it's still a new fangled innovation to me, but if that's what the kids are doing these days, i will do it.

Reply:
See: http://guides.rubygems.org/specification-reference/#required_ruby_version